### PR TITLE
First POC of example file format... fun times

### PR DESCRIPTION
**Pull request by tmeasday**
Issue: #4848

This branch is a proof-of-concept / experimentation with the example format as documented [here](https://gist.github.com/tmeasday/4de20eab47226a870ab1025642ba848c) and discussed in #4848.

## What I did

1. Added a `importAllExamples` that allows you to use the example format from https://gist.github.com/tmeasday/4de20eab47226a870ab1025642ba848c in official example.

2. Converted the first example over :)

## How to test

Run official storybook for now

### CODE REVIEW

**Comment by tmeasday** [comment]
Please feel free to push to this branch as you make further updates moving official stories over to the example format. Keep a record of whatever feels weird and I encourage as many people as possible to have a go to get a feel for it.

I've assigned a few folks I know are very interested.

File: examples/official-storybook/stories/parameters.examples.js [hunk 1]
```diff
@@ -0,0 +1,26 @@
...
+  title: 'Core|Parameters',
+  parameters: {
+    chapterParameter,
+  },
+};
+
+// I'm not sure what we should recommend regarding propTypes? are they a good idea for examples?
+// Given we sort of control the props, should we export a prop type?
+// eslint-disable-next-line react/prop-types
+export const passed = ({ parameters: { fileName, ...parameters } }) => (
```
Comment (igor-dv) [inline examples/official-storybook/stories/parameters.examples.js]: need to think now what to do with the `addon-storysource` 🤔

Comment (ndelangen) [reply examples/official-storybook/stories/parameters.examples.js]: indeed

**Comment by tmeasday** [comment]
Closing this in favour of $5058 -- if folks are interested they can migrate over stories in the `tech/overhaul-ui` branch or (even better) write new stories there!

**Pull request closed without merging**